lib/protobuf/generators/field_generator.rb

```diff
@@ -145,6 +169,27 @@ def verbatim_default_value
         descriptor.default_value
       end
 
+      def determine_type_name(descriptor)
+        case descriptor.type.name
+        when :TYPE_MESSAGE, :TYPE_ENUM, :TYPE_GROUP then
+          modulize(descriptor.type_name)
+        else
+          type_name = descriptor.type.name.to_s.downcase.sub(/type_/, '')
+          ":#{type_name}"
+        end
+      end
+
+      def determine_map_entry
+        return nil if @enclosing_msg_descriptor.nil?
+        return nil unless descriptor.label.name == :LABEL_REPEATED && descriptor.type.name == :TYPE_MESSAGE
+        # find nested message type
+        name_parts = descriptor.type_name.split(".")
+        return nil if name_parts.size < 2 || name_parts[-2] != @enclosing_msg_descriptor.name
+        nested = @enclosing_msg_descriptor.nested_type.find { |e| e.name == name_parts[-1] }
+        return nested if !nested.nil? && nested.options.try(:map_entry?) { false }
```

Comment:
`try(:map_entry?)` will return nil if it doesn't exist anyway right? So `try(:map_entry?) { false }` would be redundant